pkg/cli/mirror/operator.go

```diff
@@ -525,21 +525,114 @@ func (o *OperatorOptions) plan(ctx context.Context, dc *declcfg.DeclarativeConfi
 	return mappings, validateMapping(*dc, mappings)
 }
 
-// validateMapping will search for bundle and related images in mapping
-// and log a warning if an image does not exist and will not be mirrored
+/*
+validateMapping will search for bundle and related images in mapping
+and log a warning if an image does not exist and will not be mirrored.
+
+# Arguments
+
+• dc: the catalog content that contains bundle and related images
+
+• mapping: the source/destination mapping to search through looking for a match based on the catalog content
+
+# Returns
+
+• error: this should only produce an error if the bundle or related images in the catalog could
+not be parsed
+*/
 func validateMapping(dc declcfg.DeclarativeConfig, mapping image.TypedImageMapping) error {
-	var errs []error
+
+	/*
+		The source image in the mapping could have been modified from the value found in the catalog
+		in some circumstances. For example, when using registries.conf we might find an image in a mirror
+		and use that for the mapping rather than the original location defined in the catalog.
+
+		The docker image format could be one of these variants:
+
+			<registry>/<namespace>/name
+			<registry>/<namespace>/name:tag
+			<registry>/<namespace>/name@digest
+			<registry>/<namespace>/name:tag@digest
+
+		Because the registry and namespace portion of the docker reference *could* be different, not
+		only do we need to check for "exact matches", we also need to attempt to find a match
+		based on the name and one of the tag / digest variants above to get a match. If none of those
+		variants match than we need to add a error message.
+	*/
+
+	// getImageList will produce a list of images formatted as described above.
+	// It returns a list for consistent ordering from most likely to least likely
+	// to be encountered, to allow for early search termination.
+	getImageList := func(sourceImage image.TypedImage) (result []string) {
```

Comment:
One of the goals of future refactoring is to reduce the quantity of anonymous functions on the code base to improve the readability.Is it possible to move to a standalone func?